File: process.py
from multiprocessing import Process, Queue
from random          import randint
from time            import asctime, time
from sys             import argv

timequeue = Queue()

def worker ( id, init ):
    start = time()
    print "Process %d :: started at %s" % ( id, asctime() )

    for i in xrange( 1, 10000000 ):
        init += ( i % 4 ) * ( i % 4 ) / ( i % 10 + 1 )

    timequeue.put( time() - start )
    print "Process %s :: stopped at %s" % ( id, asctime() )

    return

proc_count = 2
try:    proc_count = int( argv[1] )
except: pass

print "Running with %d processes" % (proc_count)
proc_count += 1 # increment for range

start = time()
procs = []

for i in range( 1, proc_count ):
    Process( target = worker, args = ( i, randint( 0, 10 ) )).start()

timesum = 0
for i in range( 1, proc_count ):
    timesum += timequeue.get()

print "Master logged %f runtime" % (time() - start)
print "Threads reported %f combined runtime" % (timesum)

File: process.pl
use strict;
use warnings;
use threads;
use threads::shared;
use Time::HiRes qw/time/;
use List::Util  qw(sum);

sub workfunc {
    my ($id, $init) = @_;

    printf "Thread %d :: started at %s\n", $id, scalar localtime;
    my $start = time;
    $init += ( $_ % 4 ) * ( $_ % 4 ) / ( $_ % 10 + 1 ) for 1 .. 10_000_000;
    printf "Thread %d :: stopped at %s\n", $id, scalar localtime;

    return time - $start;
}

my $num_threads = shift || 2;
print "Running with $num_threads threads\n";

my $master_start = time;
my @threads =
  map { threads->create( \&workfunc, $_, rand 10 ); } 1 .. $num_threads;

my $thread_time = sum map { $_->join() } @threads;
my $master_time = time - $master_start;

printf "Master logged %.3f runtime\n", $master_time;
printf "Threads reported %.3f combined runtime\n", $thread_time;
